### DISCUSSION TITLE
type.register from module type error: Type EXE is not registered.

### DISCUSSION THREAD

**Discussion by nghia1303** [bfgroup/b2#418] [Q&A] [answered]
After I ran bootstrap.bat and then b2 inside my boost root directory, I receive this error. I cannot find the cause for this error. Anyone has idea about this problem? I'm trying to build boost library with default toolset is msvc 143

`D:\[02]Codes\Assignment\dependencies\boost_1_86_0\boost_1_86_0>b2
D:/[02]Codes/Assignment/dependencies/boost_1_86_0/boost_1_86_0/tools/build/src/build\type.jam:56: in type.register from module type
error: Type EXE is not registered.
D:/[02]Codes/Assignment/dependencies/boost_1_86_0/boost_1_86_0/tools/build/src/tools\stage.jam:560: in import from module stage
D:/[02]Codes/Assignment/dependencies/boost_1_86_0/boost_1_86_0/tools/build/src/tools\builtin.jam:27: in import from module builtin
D:/[02]Codes/Assignment/dependencies/boost_1_86_0/boost_1_86_0/tools/build/src\build-system.jam:12: in module scope from module build-system`

**Comment by grisumbras** [comment]
It really shouldn't happen.

[This line in `stage.jam`](https://github.com/boostorg/build/blob/boost-1.86.0/src/tools/stage.jam#L28) ensures built-in types are registered.

[This line in `types/exe.jam`](https://github.com/boostorg/build/blob/boost-1.86.0/src/tools/types/exe.jam#L7) registers `EXE` type.

Can you check that you have these lines in your Boost sources?

**Reply by nghia1303** [reply, marked as answer]
Thanks for reply, I fixed the issue by install the prebuilt version(.msi) and then change my boost location, have to ensure that the path does not contain anything except alphabets. The problem might be the square brackets in my path while running b2.

**Comment by grisumbras** [comment]
That's actually a very interesting observation!
